src/org/exist/storage/BrokerPools.java

```diff
@@ -0,0 +1,319 @@
+                //Add it to the list
+                instances.put(instanceName, instance);
+            } catch(final Throwable e) {
+                // Catch all possible issues and report.
+                LOG.error("Unable to initialize database instance '" + instanceName + "': " + e.getMessage(), e);
+                final EXistException ee;
+                if(e instanceof EXistException) {
+                    ee = (EXistException)e;
+                } else {
+                    ee = new EXistException(e);
+                }
+                throw ee;
+            }
+        } finally {
+            writeLock.unlock();
+        }
+    }
+
+    /**
+     * Returns whether or not the default database instance is configured.
+     *
+     * @return <code>true</code> if it is configured
+     */
+    public static boolean isConfigured() {
+        return isConfigured(DEFAULT_INSTANCE_NAME);
+    }
+
+    /**
+     * Returns whether or not a database instance is configured.
+     *
+     * @param instanceName The name of the database instance
+     * @return <code>true</code> if it is configured
+     */
+    public static boolean isConfigured(final String instanceName) {
+        final Lock readLock = instancesLock.readLock();
+        readLock.lock();
+        try {
+            final BrokerPool instance = instances.get(instanceName);
+            if (instance == null) {
+                return false;
+            } else {
+                return instance.isInstanceConfigured();
+            }
+        } finally {
+            readLock.unlock();
+        }
+    }
+
+    /**
+     * Returns the broker pool for the default database instance (if it is configured).
+     *
+     * @return The broker pool of the default database instance
+     *
+     * @throws EXistException If the instance is not available (not created, stopped or not configured)
+     */
+    public static BrokerPool getInstance() throws EXistException {
+        return getInstance(DEFAULT_INSTANCE_NAME);
+    }
+
+    /**
+     * Returns a broker pool for a database instance.
+     *
+     * @param instanceName The name of the database instance
+     * @return The broker pool
+     *
+     * @throws EXistException If the instance is not available (not created, stopped or not configured)
+     */
+    public static BrokerPool getInstance(final String instanceName) throws EXistException {
+        //Check if there is a database instance with the same id
+        final Lock readLock = instancesLock.readLock();
+        readLock.lock();
+        try {
+            final BrokerPool instance = instances.get(instanceName);
+            if (instance != null) {
+                //TODO : call isConfigured(id) and throw an EXistException if relevant ?
+                return instance;
+            } else {
+                throw new EXistException("Database instance '" + instanceName + "' is not available");
```

Comment:
@adamretter I did add `instancesInitializtionException` by 7020234bb13dfd6ea933f9d99a1bebbb9f4e205c to get more informative exception. Now, you are saying that it's not a case anymore because it was 5 years ago, why?